csharp/ql/lib/ext/System.IO.model.yml

```diff
@@ -7,11 +7,24 @@ extensions:
       - ["System.IO", "File", False, "Create", "", "", "ReturnValue", "file-write", "manual"]
       - ["System.IO", "File", False, "CreateText", "", "", "ReturnValue", "file-write", "manual"]
       - ["System.IO", "File", False, "Open", "", "", "ReturnValue", "file-write", "manual"]
+      - ["System.IO", "File", False, "Open", "", "", "ReturnValue", "file", "manual"]
+      - ["System.IO", "File", False, "OpenRead", "", "", "ReturnValue", "file", "manual"]
+      - ["System.IO", "File", False, "OpenText", "", "", "ReturnValue", "file", "manual"]
```

Comment:
I can do that if we think it's beneficial. I used the `""` signatures since that was the norm for the `file-write` models.

Reply:
@michaelnebel What's the general guidance on specifying signatures?

Reply:
Good question!
This is not something that has been discussed and the following just reflects my opinion: I would choose not to include the signature, if it looks "reasonable" that the model won't suddenly be incorrect (that is, primarily use empty signatures when it is summaries that applies to eg. `Argument[0]` and `ReturnValue` - where it generally will be unlikely that an overload will be introduced where the model will become incorrect for that overload).
From a performance perspective, it will perform slightly better, if we use empty signatures in case they match multiple callables as this limits the amount of text that needs to be read from the YAML files before query execution, but I don't think this will be measurable in any way.
@tamasvajk : Do you prefer that we include the full signature?

Reply:
Let's keep them as they are. When we have a general guideline, we can revisit these APIs if they don't match the guidance.

Reply:
It is a valid point that it will look like we have "less" summaries if we use the wildcard patterns. The same also goes if we set the `subtypes` to `true`. I hope that our success on this area is no longer measured by the metric that counts the number of models.

Reply:
Also for the sources modelled here, it seems very likely that it will actually help us to use an empty signature as it is very likely that if yet another overload is introduced then it is also a source.